src/apis/brain.js

```diff
@@ -633,69 +633,41 @@ export const withdrawInvestorProfit = async (user, asset, notificationId, update
     }
   });
 
-export const fundAsset = async (user, assetId, amount, onFailureContributionPopup, onSuccessRefreshData, notificationId, assetName, updateNotification, amountDollars) =>
-  new Promise(async (resolve, reject) => {
-    try {
-      updateNotification(notificationId, {
-        metamaskProps: {
-          assetName: assetName,
-          operationType: 'contribution',
-        },
-        status: 'info',
+export const fundAsset = async (userName, assetId, amount, onTransactionHash, onReceipt, onError) => {
+  try {
+    const fundingHubContract = new window.web3js.eth.Contract(
+      FundingHub.ABI,
+      FundingHub.ADDRESS,
+    );
+    const weiAmount = window.web3js.utils.toWei(amount.toString(), 'ether');
```

Comment:
Really? Does it take Strings as a parameter?

Reply:
yes, either string or BigNumber

Reply:
Jesus.